protostar/migrator/cheatcodes/migrator_invoke_cheatcode.py

```diff
@@ -14,15 +20,39 @@
 from protostar.utils.data_transformer import CairoOrPythonData
 
 
+class SignedCheatcodeConfig(CheatcodeNetworkConfig):
+    max_fee: NotRequired[int]  # In Wei
+    auto_estimate_fee: NotRequired[bool]
+
+
+@dataclass
+class ValidatedSignedCheatcodeConfig:
+    max_fee: Optional[int] = None
+    wait_for_acceptance: bool = False
+    auto_estimate_fee: bool = False
+
+    @classmethod
+    def from_dict(
+        cls, config: Optional[SignedCheatcodeConfig]
+    ) -> "ValidatedSignedCheatcodeConfig":
+        if not config:
+            return cls()
+        return cls(
+            wait_for_acceptance=config.get("wait_for_acceptance", False),
+            max_fee=config.get("max_fee", None),
+            auto_estimate_fee=config.get("auto_estimate_fee", False),
+        )
```

Comment:
Maybe we should also typecheck the values here and raise errors eagerly? Wdyt?

Reply:
That's not necessary, since have type safety from typeddict. End user error is possible, but type-check step in CI won't pass then.

Reply:
oh didn't notice that. great!